Update CreateDemoUsers.ps1

KhaaliTurbo opened the pull request:
First names were never random; I only got the first name consistently. The last name wasn't totally random either so might as well use the same script.

KhaaliTurbo committed:
Update CreateDemoUsers.ps1

First names were never random; I only got the first name consistently. The last name wasn't totally random either so might as well use the same script.

semyonsh commented:
Considerably slows down script, is there a way to optimize this solution to make it faster?

KhaaliTurbo commented:
I made some changes, and also some more. I should be able to update it next week.

KhaaliTurbo committed:
Update CreateDemoUsers.ps1

- added $DebugPreference (see below as to why);
- prepared $phoneCountryCodes for more countries. Now loads into an array list;
- updated comment for $locationCount;
- built in a check for $locationCount amount against $phoneCountryCodes amount;
- slightly tuned the import of the names and their character readability;
- import the first- and last names in to an arraylist;
- capitalise first letter of first- and last name;
- match the phone country code against the selected country;
- dump output of ERRORn to debug mode (Continue), but there aren't any now since the matchcc method is now introduced. There will only be errors if the Addresses and PostalAreaCode files aren't staged correctly (will update these momentarily);
- $officePhone country code now matches the country;
- added the $officePhone and $country to the "Created user #" output for monitoring and verification.

KhaaliTurbo committed:
Supplements $phoneCountryCodes' NL and DE countries

KhaaliTurbo reviewed (commented):
Just noticed; some cities aren't matching yet to their respective country, but that's easily fixed in an upcoming release. The code is already there vis-á-vis the added $matchcc method.
